cpp/grammar_matcher.cc

```diff
@@ -270,7 +270,9 @@ class GrammarMatcher::Impl : public GrammarMatcherBase {
 
   bool AcceptString(const std::string& input_str, bool debug_print = false);
 
-  bool FillNextTokenBitmask(DLTensor* next_token_bitmask, int index, bool debug_print = false);
+  bool FillNextTokenBitmask(
+      const DLTensor& next_token_bitmask, int index, bool debug_print = false
```

Comment:
The DLTensor* input mainly indicates the function will "modify" the content of the tensor. It's a convention in XGrammar that all functions modifying their input will pass by pointer. So I think we can keep DLTensor* here.